packages/svelte/src/compiler/phases/2-analyze/index.js

```diff
@@ -1052,6 +1058,33 @@ const function_visitor = (node, context) => {
 	});
 };
 
+/**
+ * A 'safe' identifier means that the `foo` in `foo.bar` or `foo()` will not
+ * call functions that require component context to exist
+ * @param {import('estree').Expression | import('estree').Super} expression
+ * @param {Scope} scope
+ */
+function is_safe_identifier(expression, scope) {
+	let node = expression;
+	while (node.type === 'MemberExpression') node = node.object;
+
+	if (node.type !== 'Identifier') return false;
+
+	const binding = scope.get(node.name);
+
+	if (
+		binding &&
+		(binding.kind === 'prop' ||
+			binding.kind === 'bindable_prop' ||
+			binding.kind === 'rest_prop' ||
+			binding.declaration_kind === 'import')
+	) {
+		return false;
```

Comment:
[also needs to handle store subscriptions](https://svelte-5-preview-3ok9g0vam-svelte.vercel.app/#H4sIAAAAAAAACj2LwQrCMBAFfyU8hCoEe49F8DuMB2k3GGiyYbNVpPTfSy49zjCzIsSZKtxzRX4ngsOjFFjovzSoX5qVYFF5kbGZoY4Si9599hpTYVETmE0QTqa79pUT_T4k1LXgFJjPl5vPQ39ssEg8xRBpglNZaHttOzX3HLiHAAAA)

Reply:
good catch, done